### PR TITLE
Stop ticking animations on non-dirty nodes during traversal

### PR DESCRIPTION
**Pull request by bholley**
See #13865.

r? @emilio

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13866)

### CODE REVIEW

File: components/script/script_thread.rs [hunk 1]
```diff
@@ -1536,6 +1536,7 @@ impl ScriptThread {
         let node = unsafe { node.get_jsmanaged().get_for_script() };
         let window = window_from_node(node);
 
+        node.dirty(NodeDamage::NodeStyleDamaged);
```
Comment (emilio) [inline components/script/script_thread.rs]: Could you point to https://github.com/servo/servo/issues/13865 here too?

**Review by emilio — COMMENTED** [review]
r=me with that

**Comment by bholley** [comment]
@bors-servo p=1

(another PR based on this one)

**Pull request merged**